src/plone/restapi/tests/test_services_vocabularies.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from plone import api
+from plone.app.testing import setRoles
+from plone.app.testing import SITE_OWNER_NAME
+from plone.app.testing import SITE_OWNER_PASSWORD
+from plone.app.testing import TEST_USER_ID
+from plone.restapi.testing import PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+from plone.restapi.testing import RelativeSession
+from zope.component import provideUtility
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleVocabulary
+from zope.schema.vocabulary import SimpleTerm
+
+import transaction
+import unittest
+
+
+TEST_TERM_1 = SimpleTerm(42, token='token1', title=u'Title 1')
+TEST_TERM_2 = SimpleTerm(43, token='token2', title=u'Title 2')
+TEST_VOCABULARY = SimpleVocabulary([TEST_TERM_1, TEST_TERM_2])
+
+
+def test_vocabulary_factory(context):
+    return TEST_VOCABULARY
+
+
+def test_context_vocabulary_factory(context):
+    return SimpleVocabulary([
+        SimpleTerm(context.id, token='id', title=context.id),
+        SimpleTerm(context.title, token='title', title=context.title)
+        ])
+
+
+class TestVocabularyEndpoint(unittest.TestCase):
+
+    layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+
+    maxDiff = None
+
+    def setUp(self):
+        self.app = self.layer['app']
+        self.portal = self.layer['portal']
+        self.portal_url = self.portal.absolute_url()
+        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+
+        self.api_session = RelativeSession(self.portal_url)
+        self.api_session.headers.update({'Accept': 'application/json'})
+        self.api_session.auth = (SITE_OWNER_NAME, SITE_OWNER_PASSWORD)
+        provideUtility(test_vocabulary_factory, provides=IVocabularyFactory,
+                       name='plone.restapi.tests.test_vocabulary')
+
+    def test_get_vocabulary(self):
+        response = self.api_session.get(
+            '/@vocabularies/plone.restapi.tests.test_vocabulary')
+
+        self.assertEqual(200, response.status_code)
+        response = response.json()
+        self.assertEqual(
+            response,
+            {u'@id': u'http://localhost:55001/plone/@vocabularies/''plone.restapi.tests.test_vocabulary',  # noqa
```

Comment:
@sneridagh I usually try not to assert a large dict to make the tests more maintainable (our testrunner sucks at showing dict diffs). Though, I guess this is a matter of taste...